content/source/docs/extend/best-practices/depending-on-providers.html.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: "extend"
+page_title: "Extending Terraform: Depending on Providers"
+sidebar_current: "docs-extend-best-practices-depending-on-providers"
+description: |-
+  How to safely depend on providers and understand their interfaces.
+---
+
+## Depending on Providers
+
+Terraform's providers are a substantial amount of code, and occasionally it
+makes sense to depend on their functionality. The most straightforward and
+obvious way to depend on a provider is to depend on the Terraform CLI, but
+occasionally it makes sense to rely on a provider in a different context.
+
+This guide lays out the supported ways to interface with and depend on
+Terraform's providers. Unless the provider explicitly states otherwise, no
+other compatibility guarantees are provided.
+
+### Importing Providers as Go Modules
+
+Terraform's providers are written as Go packages, and they mostly use Go
+modules as their dependency management solution. This makes it tempting to
+import the provider as a dependency of your Go code, and to call its exposed
+interface. This is explicitly an **unsupported** way to interact with providers
+and provider maintainers make no guarantees around backwards compatibility or
+the continued functioning of code that does this.
+
+Providers are unable to be imported as Go modules reliably because their
+versioning scheme is intended to convey information about the Terraform config
+interface the provider presents. It's unable to capture both the configuration
+interface _and_ the Go API interface in a useful way, as what is compatible in
+one may not be compatible in the other. Rather than give the impression that
+the package should be imported by using the `/vX` suffix now mandated for
+versions after 2.0.0, providers have chosen to make their incompatibility with
+being imported into Go code explicit.
+
+If you find yourself needing to do this, perhaps one of the methods below will
+work for you, and is explicitly supported and covered under versioning policies.
+If not, please reach out and [open an issue](https://github.com/hashicorp/terraform/issues/new)
+and we'll try to come to an appropriate solution.
```

Comment:
I would be more explicit and try to nudge them to come up with a specific use case here, if possible. I mean - correct me if I'm wrong but it's unlikely that we make a provider importable just because someone asks for it, but it's more likely that we'd come up with a solution to a problem that doesn't involve importing providers.

Reply:
Pushed a quick change! Let me know if that addresses your concern.

Reply:
That new wording is perfect! :shipit: